crates/services/consensus_module/poa/src/service_test.rs

```diff
@@ -148,21 +152,42 @@ impl TestContextBuilder {
 
         let p2p_port = generate_p2p_port();
 
+        let signer = FakeBlockSigner;
+
         let service = new_service(
             &BlockHeader::new_block(BlockHeight::from(1u32), Tai64::now()),
             config,
             txpool,
             producer,
             importer,
             p2p_port,
+            signer,
         );
         service.start().unwrap();
         TestContext { service }
     }
 }
 
+struct FakeBlockSigner;
```

Comment:
Looks like both the trait methods are both unimplemented. That's a smell to me.

Of course, it's there because we need to provide some `impl` for the trait, but maybe we should have tests that actually use this?

If not, then maybe we could use the `Unavailable` variant here?

Reply:
Yep and it was also causing tests to fail. I had just forgotten that I actually have to implement those. Done in 225cfc3.